tests/parser/array-destructuring.ts

```diff
@@ -3,4 +3,21 @@ function test1(): void {
   const {d, e} = f;
   const [] = g;
   const [,] = h;
+
+  const [,,  ,
+    ,] = k;
+
+  let [
+  ] = n;
+
+  var [,_,] = j ;
+  var [,,  ,a,, f ,
+    , g,  g,        ] = 1;
+  var [n, a, a, a,v,
+    ,h,h,h,h,h,,r
+    ,,r,,h
+    ,y, ] = 2;
+  let {aaa , a} = 1;
+  var {f,,,,,,,,,,,,} = 3;
+  const {fff} = 3;

```

Comment:
Btw all that should an error like:
```
TS2488: Type 'number' must have a iterator.
```

Reply:
Yes, but it's a parser test, not a compiler test

Reply:
If this is a request I'll add a test for that but it's unclear what you're asking :)